spec/requests/groups_controller_spec.rb

```diff
@@ -0,0 +1,108 @@
+require 'rails_helper'
+
+RSpec.describe 'Groups', type: :request do
```

Comment:
I think the integration test choice here is the right one. If we want, we can add further capybara tests to make sure elements on the page are rendering as we expected, but for the time being I think this covers things well.